packages/pds/src/sequencer/sequencer.ts

```diff
@@ -133,35 +156,65 @@ export class Sequencer extends (EventEmitter as new () => SequencerEmitter) {
     return seqEvts
   }
 
-  async pollDb() {
+  private async pollDb(): Promise<void> {
+    if (this.destroyed) return
+    // if already polling, do not start another poll
     try {
-      this.polling = true
       const evts = await this.requestSeqRange({
         earliestSeq: this.lastSeen,
         limit: 1000,
       })
       if (evts.length > 0) {
-        this.queued = true // should poll again immediately
+        this.triesWithNoResults = 0
         this.emit('events', evts)
         this.lastSeen = evts.at(-1)?.seq ?? this.lastSeen
+      } else {
+        this.triesWithNoResults++
+        // when no results, exponential backoff on pulling, with a max of a second wait
+        const waitTime = Math.min(Math.pow(2, this.triesWithNoResults), SECOND)
+        await wait(waitTime)
       }
+      this.pollPromise = this.pollDb()
     } catch (err) {
       log.error({ err, lastSeen: this.lastSeen }, 'sequencer failed to poll db')
```

Comment:
I could imagine treating the error case as being a "try with no results."  It might be a little safety mechanism in case it's constantly failing and spinning on this.

Reply:
good call switched that out 👍